rfcs/text/remove-support-for-uncontrolled-inputs.md

```diff
@@ -0,0 +1,232 @@
+
+# Alternatives
+
+Do nothing; we leave the input components as they are and continue to support controlled and uncontrolled behaviours. This course of action would mean no impact to our consumers. Uncontrolled input components can be used with Formik, even though this is not encouraged, through implementing a store type (React's [useReducer hook](https://reactjs.org/docs/hooks-reference.html#usereducer) or [Redux](https://redux.js.org/) for example) mechanism in order to retrieve a value. However, we would always still recommend using controlled input components with Formik.
+
+# Adoption strategy
+
+This proposed work will be implemented as a `BREAKING CHANGE`. In terms of how consuming projects will upgrade Carbon, we will need to document the upgrade process and link to it in the release notes. If a project has used an input as uncontrolled, they will need to update their codebase to make this controlled instead. This will involve making sure that the value is stored using React's [useState](https://reactjs.org/docs/hooks-state.html) hook to retain the value and making sure this is managed using an `onChange` function. We may have to provide additional support in this scenario to provide examples of how they can achieve the same functionality using this approach.
+
+We also already provide various codemods in our [Carbon Codemod](https://github.com/Sage/carbon-codemod) repo that could be used by our consumers to assist with the move to a controlled component. Examples of this are:
+- A [rename-prop](https://github.com/Sage/carbon-codemod/tree/master/transforms/rename-prop) codemod that consumers could use to change a `defaultValue` value prop to `value`.
+- A [add-prop](https://github.com/Sage/carbon-codemod/tree/master/transforms/add-prop) codemod that could be used to add to a `value` prop.
+- A [remove-prop](https://github.com/Sage/carbon-codemod/tree/master/transforms/remove-prop) codemod that could be used to remove a `defaultValue` prop.
+- A [replace-prop-value](https://github.com/Sage/carbon-codemod/tree/master/transforms/replace-prop-value) codemod that changes the value of a prop.
+
+There is also support for writing controlled components in React's documentation. This can be found [here](https://reactjs.org/docs/forms.html#controlled-components).
+
+# How we teach this
+
+This RFC will be communicated in the Sage `#carbon` and `#carbon-announcement` slack channels, where we will ask our consumers to feedback on the RFC. Beyond that, it will be visible in the release notes when consumers upgrade. This RFC document will reside in the repository for future reference.
+
+We will be able to support consumers of Carbon in the `#carbon` slack channel too. 
+
+# Documentation
+
+As part of this work we will ensure that Storybook examples are updated where any uncontrolled behaviours are demonstrated. This will involve removing any uncontrolled examples and ensuring we have the controlled examples.
```

Comment:
We should also be documenting that we don't support uncontrolled usage in these components otherwise we'll get issues raised about it.

Reply:
Good plan. I'll add something in here that we'll specifically state in the component description that we only support controlled usage.